fe/fe-core/src/main/java/com/starrocks/catalog/MaterializedView.java

```diff
@@ -1353,6 +1353,9 @@ private Set<String> getPartitionedMVRefreshPartitions(boolean isQueryRewrite) {
             }
             // If the non ref table has already changed, need refresh all materialized views' partitions.
             Set<String> partitionNames = getUpdatedPartitionNamesOfTable(baseTable, true, isQueryRewrite);
+            // NOTE: to check whether the existing partition is updated
+            Set<String> mvPartitionsNames = getRangePartitionMap().keySet();
+            partitionNames.retainAll(mvPartitionsNames);
```

Comment:
partitionNames is the changed partitions of base table, but mvPartitionsNames is the partition names of mv.
I think we can not decide the relationship between them

Reply:
I think you should consider a.id_date = b.id2_date